src/Tasks/ManifestUtil/ManifestReader.cs

```diff
@@ -250,6 +246,27 @@ private static Manifest Deserialize(Stream s)
                 return m;
             }
         }
+
+#if NET
+        private static void ReadExactly(Stream stream, byte[] buffer, int offset, int count)
+        {
+            stream.ReadExactly(buffer, offset, count);
+        }
+#else
+        private static void ReadExactly(Stream stream, byte[] buffer, int offset, int count)
```

Comment:
Why not just make this an extension method on `Stream` that is `if !NET` gated? That way you can just use `stream.ReadExactly` in the code base and get the correct behavior.

Reply:
You're absolutely right! This is a clean and elegant solution. I'll add a new extension method to `Stream` for .NET FX version.

Reply:
Updated: https://github.com/dotnet/msbuild/blob/d8efe63fdc2c38a04d1fb1be7fc99d807453b664/src/Tasks/ManifestUtil/ManifestReader.cs#L269-L302